### DISCUSSION TITLE
Leveraging e-Signet and INJI from MOSIP

### DISCUSSION THREAD

**Discussion by surendrasinghs** [Sunbird-RC/community#352] [Ideas & Enhancements]
Folks, I came across two (out of many) interesting components in MOSIP that we (the RC community) could leverage and fast-track some of the developments.

1. e-Signet: simple yet powerful mechanism for end users to identify themselves to avail of online services and also share their profile information. It supports multiple modes of identity verification. Check out more at https://docs.esignet.io/ and https://docs.mosip.io/1.2.0/integrations/e-signet
2. INJI (Resident Mobile Application): Mobile application to securely store, share and receive digital credentials. Check out https://docs.mosip.io/1.2.0/modules/mobile-application and https://github.com/mosip/inji

From my understanding, Sunbird RC can be used as an 'authentication registry' using which users can log in (authenticate themselves) in other systems. I think e-Signet already solves this and more. Would love to hear thoughts on possibility of leveraging and collaborating @tejash-jl @ChakshuGautam @dileepbapat @coolbung @parthlawate @suresh12 @sukhpreetsamagra @rahul101001000 @amarts @girirajdaga @pramodkvarma 

P.S. Also check out the webinar https://www.youtube.com/watch?v=W3K2I4O08vc

**Comment by sankarshanmukhopadhyay** [comment]
> From my understanding, Sunbird RC can be used as an 'authentication registry' using which users can log in (authenticate themselves) in other systems. I think e-Signet already solves this and more. 

Thanks for sharing the links! I'll be reading through the flow in e-Signet. From your description it seems closer to the classical federated design pattern of handling identity and authentication.

**Comment by rahul101001000** [comment]
Thanks @surendrasinghs, great pointers! 

My understanding is that e-Signet = eKYC as a service. It is not a VC issuer, neither is it an authenticator. So we will have (a) RC as the credential issuer, (b) auth for end users using Aadhar/Keycloak/Auth0 or Aries/ION, (c) e-Signet as a eKYC service

INJI is a perfect mobile wallet app equivalent to LearnerCard, Affinidi, et al. It is still different than a Digilocker or the RC based cloud wallet. We still need that cloud wallet where all creds are stored per user. 

Can anyone confirm my understanding or please help correct it. 

Thanks in advance!

**Reply by pramodkvarma** [reply]
That's my understanding as well @rahul101001000

**Comment by surendrasinghs** [comment]
Thanks @rahul101001000 for the response. Please help us arrive at the decision (see below).

I am trying to narrow down to the decision of whether RC can benefit from e-Signet (I may have assumed that e-Signet solves a problem that we are trying/want to solve as well). Here's my thought process:

Use-cases that I was visualizing
1. I can sign in to any other system using my account created on a Sunbird RC system - already possible ✅. 
   1. So now other systems don't need an account creation, management, and authentication system, they can use this. 
   2. But here's a caveat that RC internally has biased implementation using KeyCloak. We can fix this.
2. When I sign in to any other system using my Sunbird RC account, I can also share my profile details for quick onboarding - already possible ✅ 
   1. We can show a consent about the fields that will be shared, and also configure what fields (non-public fields) can be shared.
5. I can sign in to a Sunbird RC system using my account in another system - not yet done, in the roadmap 🟡 
   1. Is there anything of value from e-Signet or shall we just fix this in RC? I believe here's where the KeyCloak implementation needs to be generalized.

Also, from my reading of the e-Signet, I gathered that it's more than eKYC service. It also enables various modes of authentication. (Referring to https://docs.esignet.io/#user-centricity). 

@dileepbapat @tejash-jl @coolbung @parthlawate @ChakshuGautam @suresh12 - let's narrow down on a decision in a day or so?

**Comment by pramodkvarma** [comment]
Bring Sasi into this thread. He can help us answer these. @surendrasinghs

**Reply by surendrasinghs** [reply]
@gsasikumar please help us here :)

**Comment by rahul101001000** [comment]
Look forward to Sasi's guidance on where exactly would Signet be most appropriate. 

But @surendrasinghs  a few comments on the use cases themselves - 

> Caution - below is based on my understanding, I could be wrong :)

- RC never has holders registered or authenticated. It has issuers that are authenticated. Issuers generate credentials about holders, credentials are stored in RC. 
- So in that sense, RC cannot really play the role of identity manager for users. 
- So why is identity manager needed - I have multiple ways of proving my identity - Aadhar, PAN, phone number, Gmail etc. I should be able to login to one place, import all of these credentials once and for all and then for subsequent logins just use any of these logins via this portal and it would authenticate me with one/all of them. This enables single authable KYC. 
- Ideally a Aries or Ion would be perfect for it. In India, we have [Meri Pehechan](https://www.digitalindiaportal.net/meri-pehchan-portal/) which seems to do something similar. Not sure how the backend works. 
- eSignet seems similar to Meri Pehechan or Aries or Ion. So if we had to add an identity manager for RC, eSignet (or Meripehechan) seems perfect. KeyCloak cannot do what eSignet does. 

--
@gsasikumar Please do correct our understanding of what use cases we should be looking at from a kyc/identity perspective and what building block would be most suited to it. 

Thanks in advance, 

Regards
Rahul

**Reply by amarts** [reply]
> So why is identity manager needed

IMO this topic needs more discussions and understanding for a true VC ecosystem/platform to work. While I understand the current identity of any holder/subject is mentioned with an URN (can be PAN/Aadhaar/DL/email) is good to get started, this binding is not machine verifiable. Hence a proof of identity which can be 'established' through signatures (private/public key) is essential. Hence DIDs play an important role.

We all should be aware that for a true Verifiable Credential, Verifiable Presentation platform / ecosystem to work, we need a good, interoperable DID solution is a must. It may not be immediate need for currently running projects, but developers working on this should have a provision for DID management in their projects.
